dig_test.go

```diff
@@ -1433,6 +1379,112 @@ func TestGroups(t *testing.T) {
 		require.Error(t, err, "failed to provide")
 		assert.Contains(t, err.Error(), "flatten can be applied to slices only")
 	})
+
+	t.Run("check providers for value groups aren't called if only they're provided", func(t *testing.T) {
```

Comment:
What are your thoughts about:
"a soft value group provider is not called when only that value group is consumed"